3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/HttpClient.scala

```diff
@@ -46,20 +48,28 @@ object HttpClient {
    * @param uri full URI to request
    * @param authUser optional username for basic auth
    * @param authPassword optional password for basic auth
+   * @param body optional request body
    * @param method HTTP method
    * @return HTTP request
    */
   def buildRequest(
     uri: String,
     authUser: Option[String],
     authPassword: Option[String],
+    body: Option[String],
     method: String = "GET"
   ): HttpRequest = {
-    val req = Http(uri).method(method)
-    if (authUser.isDefined || authPassword.isDefined) {
-      req.auth(authUser.getOrElse(""), authPassword.getOrElse(""))
-    } else {
-      req
-    }
+    val req: HttpRequest = Http(uri).method(method)
+    req.maybeAuth(authUser, authPassword).maybePostData(body)
+  }
+
+  implicit class RichHttpRequest(request: HttpRequest) {
+
+    def maybeAuth(user: Option[String], password: Option[String]): HttpRequest =
+      if (user.isDefined || password.isDefined) request.auth(user.getOrElse(""), password.getOrElse(""))
+      else request
+
+    def maybePostData(body: Option[String]): HttpRequest =
+      body.map(data => request.postData(data).header("content-type", "application/json")).getOrElse(request)
```

Comment:
should the content type be an additional parameter?

Reply:
Hey @BenFradet. Holidays are always welcome! Well I didn't parameterised  `content-type` because it is always going to have a json body.